### ISSUE TITLE
Auto-update doesn't work with admins (and maybe normals)

### ISSUE THREAD

**Issue by SCdF** [medic/medic-webapp#2509]
When I `grunt dev` changes into couchdb, and I'm logged in as an admin, I don't get notified that there are changes and I should reload my browser. I have not yet tested if this is also happening with normal non admin users.

**Comment by SCdF** [comment]
I'm reasonably sure that it's [this line](https://github.com/medic/medic-webapp/blob/develop/static/js/controllers/inbox.js#L804):
```js
        Changes({
          key: 'inbox-ddoc',
          filter: function(change) {
            return change.id === '_design/medic-client';
          },
          callback: function() {
```

Because we're looking for changes in `_design/medic-client`, and perhaps that document is not actually changing? It is certainly not the document that contains the JS etc.

@garethbowen should this be changed back to `_design/medic`?

**Comment by SCdF** [comment]
(tested, reverting it back to `_design/medic` makes it work again)

**Comment by garethbowen** [comment]
This change will break the update notification for non-admins because they don't have a `_design/medic` to check. My first thought is to change the line to check for either, eg:

```js
return change.id === '_design/medic' || change.id === '_design/medic-client';
```

This would certainly be an improvement, however it doesn't entirely solve the problem. In api we check and only update the `medic-client` ddoc [if it's actually changed](https://github.com/medic/medic-api/blob/develop/ddoc-extraction.js#L28-L44) to avoid clients having to replicate the ddoc unnecessarily. Unfortunately this means if the inbox.js file is updated the medic-client ddoc won't be updated and the client will never find out (until they reload).

In retrospect I don't think we need to check whether the ddoc has actually changed - in production it'll be highly unlikely that the medic ddoc will change but the medic-client ddoc won't as client views or app_settings update frequently.

A gold plated solution might be to have a new doc, something like this...

```js
{
  _id: "appcache-version",
  version: "# rev 669 - 2016-07-14T10:05:36.089Z" // some identifier, either pulled out of the manifest.appcache line 2, or the hash of the contents or whatever...
}
```

Then when a ddoc change is detected check if we need to update that doc. Then replicate that doc to the client and change the listener to check for it. This is kind of nice because it's sending the minimal amount of data to notify of an appcache change, which is exactly what we're trying to look for.

1. make the change to listen for changes on either ddoc as above.
2. either:
  a. update medic-client ddoc on any medic ddoc change, or
  b. create the new appcache-version doc

**Comment by SCdF** [comment]
@garethbowen would it be horrid if we remotely watched another changes feed for changes just to `_design/medic`? My reasoning being:
 - You can only get webapp updates when you're online, so the feed will always work
 - It won't be high in bandwidth because you're not sending the doc data back, nor high in cpu as you're not filtering the feed (just only watching one doc)
 - It's the "most correct" because it's literally watching and basing its decisions on the thing that contains the things it cares about, and thus the most "self-documenting" (we're watching this thing because it's where the data we cares about comes from) and the most future proof

**Comment by garethbowen** [comment]
Yeah, I think having double the number of persistent connections would really limiting our server scalability, which is why I was too eager to remove the WatchDesignDoc service which did pretty much that. That's why I mentioned changing a minimal doc so we can notify clients without creating a new connection.

TL;DR: I think the additional bandwidth of sending a tiny doc to every client is much better than maintaining an additional connection per client.

**Comment by garethbowen** [comment]
### Code review

The ddocExtraction code now inspects the attached appcache manifest to see if it's updated and bumps a doc with id `appcache` if it is. This is replicated to the client which has a change listener to prompt the user to refresh on change.

**Comment by SCdF** [comment]
Nice! Minor questions on commits, but mostly for my learning.
